ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -437,6 +439,64 @@ data PeerSelectionState peeraddr peerconn = PeerSelectionState {
      }
   deriving Show
 
+-----------------------
+-- Debug copy of Peer Selection State
+--
+-- Used for dumping the peer selection state upon getting a USR1 signal.
+--
+data DebugPeerSelectionState peeraddr = DebugPeerSelectionState {
+       dpssTargets                     :: !PeerSelectionTargets,
+       dpssLocalRootPeers              :: !(LocalRootPeers peeraddr),
+       dpssPublicRootPeers             :: !(Set peeraddr),
+       dpssBigLedgerPeers              :: !(Set peeraddr),
+       dpssKnownPeers                  :: !(KnownPeers peeraddr),
+       dpssEstablishedPeers            :: !(Set peeraddr),
+       dpssActivePeers                 :: !(Set peeraddr),
+       dpssPublicRootBackoffs          :: !Int,
+       dpssPublicRootRetryTime         :: !Time,
+       dpssInProgressPublicRootsReq    :: !Bool,
+       dpssBigLedgerPeerBackoffs       :: !Int,
+       dpssBigLedgerPeerRetryTime      :: !Time,
+       dpssInProgressBigLedgerPeersReq :: !Bool,
+       dpssInProgressPeerShareReqs     :: !Int,
+       dpssInProgressPromoteCold       :: !(Set peeraddr),
+       dpssInProgressPromoteWarm       :: !(Set peeraddr),
+       dpssInProgressDemoteWarm        :: !(Set peeraddr),
+       dpssInProgressDemoteHot         :: !(Set peeraddr),
+       dpssInProgressDemoteToCold      :: !(Set peeraddr),
+       dpssUpstreamyness               :: !(Map peeraddr Int),
+       dpssFetchynessBlocks            :: !(Map peeraddr Int)
+} deriving Show
```

Comment:
Tracing types have be lazy (that's a requirement of the new tracing system). Most of the computations done in `makeDebugPeerSelectionState` are trivial so it doesn't matter much for performance.